.gitignore

```diff
@@ -0,0 +1,51 @@
+# Compiled source #
+###################
+*.com
+*.class
+*.dll
+*.exe
+*.o
+*.so
+*.pyc
+
+# Packages #
+############
+# it's better to unpack these files and commit the raw source
+# git has its own built in compression methods
+*.7z
+*.dmg
+*.gz
+*.iso
+*.jar
+*.rar
+*.tar
+*.zip
+
+# Logs and databases #
+######################
+*.log
+*.sql
+*.sqlite
+
+# LaTeX Generated Files #
```

Comment:
Yes:  it's not an issue of boilerplate, it's an issue of surprising users who might actually want to commit those things.  In general, it's better to tread lightly with default `.gitignores`.  :-)